lib/spear/connection.ex

```diff
@@ -44,25 +44,25 @@ defmodule Spear.Connection do
 
   @impl GenServer
   def handle_call({:request, request}, from, state) do
-    case request_and_stream_body(state, request) do
-      {:ok, state, request_ref} ->
-        state = put_in(state.requests[request_ref], %{from: from, response: %{}})
-
+    case request_and_stream_body(state, request, from) do
+      {:ok, state} ->
         {:noreply, state}
 
       {:error, state, reason} ->
         {:reply, {:error, reason}, state}
     end
   end
 
-  defp request_and_stream_body(state, request) do
+  defp request_and_stream_body(state, request, from) do
```

Comment:
there's a slight refactor in this module (and specifically this func) which changes the shape of the state struct `%__MODULE__{}` from having a `:responses` field (map) to having a `:requests` field (struct) which has a sub-field of `:responses` (struct)

this is because requests may be completed asynchronously (w.r.t. the server, the client still blocks) if they are larger than a single HTTP2 window, so we need to store some information about the requests, not just response information